packages/sources/kaiko-test/src/endpoint/trades.ts

```diff
@@ -0,0 +1,151 @@
+import { PriceEndpoint } from '@chainlink/external-adapter-framework/adapter'
+import { RestTransport } from '@chainlink/external-adapter-framework/transports'
+import { DEFAULT_INTERVAL, DEFAULT_MILLISECONDS, DEFAULT_SORT } from '../config'
+import { AdapterError } from '@chainlink/external-adapter-framework/validation/error'
+import { customSettings } from '../config'
+
+const inputParameters = {
+  base: {
+    aliases: ['from', 'coin'],
+    required: true,
+    type: 'string',
+    description: 'The symbol of symbols of the currency to query',
+  },
+  quote: {
+    aliases: ['to', 'market'],
+    required: true,
+    type: 'string',
+    description: 'The symbol of the currency to convert to',
+  },
+  interval: {
+    required: false,
+    description:
+      'The time interval to use in the query. NOTE: Changing this will likely require changing `millisecondsAgo` accordingly',
+    default: '2m',
+  },
+  millisecondsAgo: {
+    required: false,
+    description:
+      'Number of milliseconds from the current time that will determine start_time to use in the query',
+    default: 86_400_000, // 24 hours
+  },
+  sort: {
+    required: false,
+    description: 'Which way to sort the data returned in the query',
+    default: 'desc',
+  },
+} as const
+
+export interface RequestParams {
+  base: string
+  quote: string
+  interval?: string
+  millisecondsAgo?: number
+  sort?: string
+}
+
+export interface ResponseSchema {
+  query: {
+    page_size: number
+    start_time: string
+    interval: string
+    sort: string
+    base_asset: string
+    sources: boolean
+    ch: boolean
+    include_exchanges: string[]
+    exclude_exchanges: string[]
+    quote_asset: string
+    data_version: string
+    commodity: string
+    request_time: string
+    instruments: string[]
+    start_timestamp: number
+  }
+  time: string
+  timestamp: number
+  data: { timestamp: number; price: string }[]
+  result: string
+  access: {
+    access_range: { start_timestamp: number; end_timestamp: number }
+    data_range: { start_timestamp: number; end_timestamp: number }
+  }
+}
+
+type EndpointTypes = {
+  Request: {
+    Params: RequestParams
+  }
+  Response: {
+    Data: {
+      result: number
+    }
+    Result: number
+  }
+  CustomSettings: typeof customSettings
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseSchema
+  }
+}
+
+const calculateStartTime = (millisecondsAgo: number) => {
+  const date = new Date()
+  date.setTime(date.getTime() - millisecondsAgo)
+  return date
+}
+
+const restEndpointTransport = new RestTransport<EndpointTypes>({
```

Comment:
Rest transport is going away so we'll block this one until we can either move to a batch endpoint (there seem to be a few options for kaiko) or have support for the more inefficient single requests